prometheus_hardware_exporter/collector.py

```diff
@@ -109,3 +112,261 @@ def fetch(self) -> List[Payload]:
+        return payloads
+
+
+class IpmiSensorsCollector(BlockingCollector):
+    """Collector for ipmi sensors data."""
+
+    ipmimonitoring = IpmiMonitoring()
+
+    @property
+    def specifications(self) -> List[Specification]:
+        """Define ipmi sensors metrics specs."""
+        return [
+            Specification(
+                name="ipmi_temperature_celsius",
+                documentation="Temperature measure from temperature sensors",
+                labels=["id", "name", "state", "unit", "event"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmi_power_watts",
+                documentation="Power measure from power sensors",
+                labels=["id", "name", "state", "unit", "event"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmi_voltage_volts",
+                documentation="Voltage measure from voltage sensors",
+                labels=["id", "name", "state", "unit", "event"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmi_current_amperes",
+                documentation="Current measure from current sensors",
+                labels=["id", "name", "state", "unit", "event"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmi_fan_speed_rpm",
+                documentation="Fan speed measure, in rpm",
+                labels=["id", "name", "state", "unit", "event"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmi_fan_speed_ratio",
+                documentation="Fan speed measure, as a percentage of maximum speed",
+                labels=["id", "name", "state", "unit", "event"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmi_generic_sensor_value",
+                documentation="Generic sensor value from ipmi sensors",
+                labels=["id", "name", "state", "unit", "event", "type"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmimonitoring_command_success",
+                documentation="Indicates if the ipmimonitoring command succeeded or not",
+                labels=[],
+                metric_class=GaugeMetricFamily,
+            ),
+        ]
+
+    def fetch(self) -> List[Payload]:
+        """Load ipmi sensors data."""
+        sensor_data = self.ipmimonitoring.get_sensor_data()
+
+        if not sensor_data:
+            logger.error("Failed to get ipmi sensor data.")
+            return [Payload(name="ipmimonitoring_command_success", labels=[], value=0.0)]
+
+        payloads = [Payload(name="ipmimonitoring_command_success", labels=[], value=1.0)]
+        for sensor_data_item in sensor_data:
```

Comment:
It's interesting that we can only identify the items from "Units" type..  
It may be wrong some day. Is there any better way?

Reply:
Another way could be identifying by the `type` field of the sensor data item. But, since the format of this data is part of standard defined by IPMI, it's unlikely it will be changed often/without notice. [I took reference from here](https://github.com/prometheus-community/ipmi_exporter/blob/master/collector_ipmi.go#L173) for this implementation.